components/log-viewer-webui/server/src/routes/query.js

```diff
@@ -1,5 +1,57 @@
-// eslint-disable-next-line no-magic-numbers
-const EXTRACT_IR_TARGET_UNCOMPRESSED_SIZE = 128 * 1024 * 1024;
+import settings from "../../settings.json" with {type: "json"};
+import {QUERY_JOB_TYPE} from "../DbManager.js";
+
+
+/**
+ * Submits a stream extraction job with the given parameters and waits for it.
+ *
+ * @param {import("fastify").FastifyInstance | {dbManager: DbManager}} fastify
+ * @param {QUERY_JOB_TYPE} extractJobType
+ * @param {string} streamId
+ * @param {int} sanitizedLogEventIdx
+ * @throws {Error} if the extract stream job submission fails.
+ */
+const submitAndWaitForExtractStreamJob = async (
+    fastify,
+    extractJobType,
+    streamId,
+    sanitizedLogEventIdx
+) => {
+    let jobConfig;
+    const streamTargetUncompressedSize = settings.StreamTargetUncompressedSize;
+    if (QUERY_JOB_TYPE.EXTRACT_IR === extractJobType) {
+        jobConfig = {
+            file_split_id: null,
+            msg_ix: sanitizedLogEventIdx,
+            orig_file_id: streamId,
+            target_uncompressed_size: streamTargetUncompressedSize,
+        };
+    } else if (QUERY_JOB_TYPE.EXTRACT_JSON === extractJobType) {
+        jobConfig = {
+            archive_id: streamId,
+            target_chunk_size: streamTargetUncompressedSize,
+        };
+    }
+
+    if (null === jobConfig) {
+        const err = new Error(`Unsupported Job type: ${extractJobType}`);
+        err.statusCode = 400;
+        throw err;
+    }
+
+    const extractResult = await fastify.dbManager.submitAndWaitForExtractStreamJob(
+        jobConfig,
+        extractJobType
+    );
+
+    if (null === extractResult) {
+        const err = new Error("Unable to extract stream with " +
+            `streamId=${streamId} at logEventIdx=${sanitizedLogEventIdx}`);
+
+        err.statusCode = 400;
+        throw err;
+    }
+};
```

Comment:
If you agree, the logics in this helper can be moved into `dbManager.submitAndWaitForExtractStreamJob`. Let's see the suggestion there.
```suggestion
```